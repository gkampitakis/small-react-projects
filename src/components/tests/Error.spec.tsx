import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import ErrorComponent from '../Error';


describe('Error Component', () => {
  it('Should render correctly', () => {
    const tree = renderer
      .create(<ErrorComponent message={'Mock Message'} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should render error message and print message to console', () => {
    const logSpy = jest.spyOn(console, 'log').mockImplementation(() => { void (0) });
    const error = Error('Mock Error')
    const { baseElement } = render(<ErrorComponent error={error} />);

    expect(baseElement).toMatchSnapshot();
    expect(logSpy).toHaveBeenCalledWith(error);
  });
});
