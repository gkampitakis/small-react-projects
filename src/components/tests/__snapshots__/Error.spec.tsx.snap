// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Component Should render correctly 1`] = `
<main
  className="error_component"
>
  <img
    alt="Error page image"
    src="./images/error.svg"
  />
  <footer>
    <h2>
      Mock Message
    </h2>
  </footer>
</main>
`;

exports[`Error Component Should render error message and print message to console 1`] = `
<body>
  <div>
    <main
      class="error_component"
    >
      <img
        alt="Error page image"
        src="./images/error.svg"
      />
      <footer>
        <h2>
          Mock Error
        </h2>
        <p>
          You can check console for more information
        </p>
      </footer>
    </main>
  </div>
</body>
`;
